dask/dataframe/core.py

```diff
@@ -5073,21 +5073,23 @@ def set_index(
         npartitions: int | Literal["auto"] | None = None,
         divisions: Sequence | None = None,
         inplace: bool = False,
+        sort: bool = True,
```

Comment:
It's unfortunate that we already have a `sorted=` kwarg that's really close to `sort`. I'm fine stick with `sort` if we can't think of a better name.

Reply:
Only going off the conversation from the linked issue, `sort` being recommended over `repartition`, but I have no strong feelings here; open to suggestions.

Reply:
If I understand correctly, `sorted=True` with `divisions=None` would result in the same behavior as `sort=False`. The difference is that `sorted=True` with provided divisions will validate and set the divisions. So, if we supported providing divisions with `sort=False`, we could deprecate `sorted=True` in favor of `sort=False`. These provided divisions would follow the same assumptions as in the case of `sorted=True`. What are your thoughts, @jrbourbeau? To avoid scope creep, I think it might make sense to tackle that in a follow-up PR.